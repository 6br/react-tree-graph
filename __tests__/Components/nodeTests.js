import React from 'react';
import { shallow } from 'enzyme';

import Node from '../../Components/node';

describe('<Node>', () => {
	test('renders correctly', () => {
		const props = {
			x: 1,
			y: 2,
			keyProp: '',
			labelProp: 'name',
			htmlProps: {
				g: {
					className: 'test'
				}
			},
			offset: 3,
			radius: 4,
			name: 'Test Node'
		};

		const tree = shallow(<Node {...props}/>);
		expect(tree).toMatchSnapshot();
	});

	test('click event has correct parameters', () => {
		const clickMock = jest.fn();
		const event = {};

		const props = {
			x: 1,
			y: 2,
			keyProp: 'id',
			labelProp: 'name',
			htmlProps: {
				g: {
					onClick: clickMock
				}
			},
			offset: 3,
			radius: 4,
			id: 'testKey',
			name: 'Test Node'
		};

		const tree = shallow(<Node {...props}/>);
		tree.find('g').simulate('click', event);

		expect(clickMock).toHaveBeenCalledTimes(1);
		expect(clickMock).toHaveBeenCalledWith('testKey', event);
	});

	test('clicking with no prop handler does nothing', () => {
		const props = {
			x: 1,
			y: 2,
			keyProp: 'id',
			labelProp: 'name',
			htmlProps: {},
			offset: 3,
			radius: 4,
			id: 'testKey',
			name: 'Test Node'
		};

		const tree = shallow(<Node {...props}/>);
		tree.find('g').simulate('click');
	});

	test('htmlProps applied to all elements', () => {
		const props = {
			x: 1,
			y: 2,
			keyProp: '',
			labelProp: 'name',
			htmlProps: {
				g: {
					className: 'g'
				},
				circle: {
					className: 'circle'
				},
				text: {
					className: 'text'
				}
			},
			offset: 3,
			radius: 4,
			name: 'Test Node'
		};

		const tree = shallow(<Node {...props}/>);
		expect(tree).toMatchSnapshot();
	});
});