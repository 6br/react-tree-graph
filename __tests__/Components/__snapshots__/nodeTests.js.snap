// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Node> renders correctly 1`] = `
<g
  className="test"
  onClick={[Function]}
  transform="translate(2, 1)"
>
  <circle
    r={4}
  />
  <text
    dx={4.5}
    dy={3}
  >
    Test Node
  </text>
</g>
`;
